sycl/test/check_device_code/no_offset.cpp

```diff
@@ -0,0 +1,46 @@
+// RUN: %clangxx -fsycl-device-only -fsycl-early-optimizations -fsycl-dead-args-optimization -D__SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING__ -S -emit-llvm -o - %s | FileCheck %s
+
+#include <CL/sycl.hpp>
+
```

Comment:
It would be nice to have E2E test in https://github.com/intel/llvm-test-suite as well. Are we planning to add it there? Basic requirement for the test to check:
1. It doesn't  crash during AOT and JIT compilation and execution on different devices (CPU/GPU/FPGA) (I assume, later the IR will be translated to SPIR-V and we need to check that this SPIR-V is acceptable for all of the mentioned devices (I'm sure, that it will be accepted, but still));
2. Basic math operations with 'no-offsetted' accessors are having the correct result.